lightningd/plugin.c

```diff
@@ -215,28 +217,53 @@ static void destroy_plugin(struct plugin *p)
 	}
 }
 
+static u32 file_checksum(const char* path)
+{
+	FILE *fp;
+	char buf[1024];
+	size_t nread;
+	u32 checksum = 0;
+
+	fp = fopen(path, "r");
+	if (fp == NULL)
+		return 0;
+	while ((nread = fread(buf, 1, sizeof(buf), fp)) > 0)
+		checksum = crc32c(checksum, &buf, nread);
+	fclose(fp);
+	return checksum;
+}
```

Comment:
I would have just used grab_file?

This will be a fun one to test: probably need to make a temp subdir under the lightning dir and copy a plug-in there?

Reply:
I thought the same about the test... which is why there is no test yet :D
... thinking ...